src/main/java/Duke.java

```diff
@@ -1,10 +1,152 @@
+     */
+    public static boolean isNumeric(String word) {
+        int valueToConvert;
+        try {
+            valueToConvert = Integer.parseInt(word);
+            return true;
+        } catch (NumberFormatException e) {
+            System.out.println("Incorrect input, please enter an integer.");
+        }
+        return false;
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        greetingMessage();
+
+        String line;
+        Scanner in = new Scanner(System.in);
+        line = in.nextLine();
+        Task[] tasks = new Task[100];
+        int taskCount = 0;
+
+        while (!line.equals("bye")) { // Exits the program if input is "bye"
+            String[] words = line.split(" ");
+            if (line.equals("list")) {
+                // List out all the tasks added
+                list(tasks, taskCount);
+            } else if (words[0].equals("unmark") && (words.length == 2) && (isNumeric(words[1]))) {
+                // Mark a task as not done
+                unmarkTask(tasks, taskCount, words);
+            } else if (words[0].equals("mark") && (words.length == 2) && (isNumeric(words[1]))) {
+                // Mark a task as done
+                markTask(tasks, taskCount, words);
+            } else {
+                // Adding a task to the list
+                addTask(line, tasks, taskCount);
+
+                // Printing out added task message
+                addedTaskMessage(tasks[taskCount], taskCount);
+                taskCount++;
+            }
+            line = in.nextLine();
+        }
+
+        // Exiting the program
+        exitMessage();
+    }
+
+    private static void addTask(String line, Task[] tasks, int taskCount) {
+        if (line.contains(" /by")) {
+            // Adding a Deadline
+            String description = line.substring(0, line.indexOf("/by")).trim();
+            String deadline = line.substring(line.indexOf("/by") + 3).trim();
+            tasks[taskCount] = new Deadline(description, deadline);
+        } else if (line.contains(" /from") && line.contains(" /to")) {
+            // Adding an Event
+            String description = line.substring(0, line.indexOf("/from")).trim();
+            String start = line.substring(line.indexOf("/from") + 5, line.indexOf("/to")).trim();
+            String end = line.substring(line.indexOf("/to") + 3).trim();
+            tasks[taskCount] = new Event(description, start, end);
+        } else {
+            // Adding a _Todo_
+            tasks[taskCount] = new Todo(line);
+        }
+    }
+
+    private static void exitMessage() {
+        System.out.println("\t____________________________________________________________");
```

Comment:
Printing of separation line is used multiple times. It would be better to define the line as a static final literal for reusability.